core/lib/server/cache/memoize.ts

```diff
@@ -93,46 +94,60 @@ export function memoize<P extends unknown[], R>(cb: Callback<P, R>, opts?: Memoi
 				revalidate: duration,
 				tags: ["all", ...revalidateTags],
 			};
-			if (logDataCache) {
-				let dataCacheMiss = false;
-				const audit = new Audit();
-				const data = await unstable_cache(
-					async () => {
-						dataCacheMiss = true;
-						return cb(...args);
-					},
-					cacheKey,
-					nextOpts
-				)();
-				const time = audit!.getSec();
-				const isSame = oldData === data;
-				logger.debug(
-					`Data Cache - ${logID}${cb.name} ${dataCacheMiss ? "MISS" : "HIT"} ${time.toPrecision(3)}s ${dataCacheMiss ? (isSame ? "background-revalidation" : "on-demand revalidation") : ""} `
-				);
-				if (logVerbose)
-					logger.info(
-						` └ Function: ${cb.name || "Anon Func"} | args: ${JSON.stringify(args)} | tags: ${nextOpts.tags} | cacheKey: ${cacheKey}`
+			try {
+				if (logDataCache) {
+					let dataCacheMiss = false;
+					const audit = new Audit();
+					const data = await unstable_cache(
+						async () => {
+							dataCacheMiss = true;
+							return cb(...args);
+						},
+						cacheKey,
+						nextOpts
+					)();
+					const time = audit!.getSec();
+					const isSame = oldData === data;
+					logger.debug(
+						`Data Cache - ${logID}${cb.name} ${dataCacheMiss ? "MISS" : "HIT"} ${time.toPrecision(3)}s ${dataCacheMiss ? (isSame ? "background-revalidation" : "on-demand revalidation") : ""} `
 					);
-				oldData = data;
-				return data;
-			} else {
-				const data = await unstable_cache(
-					async () => {
-						return cb(...args);
-					},
-					[cb.toString(), JSON.stringify(args), ...additionalCacheKey],
-					{
-						revalidate: duration,
-						// we always cache "all"
-						tags: ["all", ...revalidateTags],
-					}
-				)();
-				return data;
+					if (logVerbose)
+						logger.info(
+							` └ Function: ${cb.name || "Anon Func"} | args: ${JSON.stringify(args)} | tags: ${nextOpts.tags} | cacheKey: ${cacheKey}`
+						);
+					oldData = data;
+					return data;
+				} else {
+					const data = await unstable_cache(
+						async () => {
+							return cb(...args);
+						},
+						[cb.toString(), JSON.stringify(args), ...additionalCacheKey],
+						{
+							revalidate: duration,
+							// we always cache "all"
+							tags: ["all", ...revalidateTags],
+						}
+					)();
+					return data;
+				}
+			} catch (error) {
+				const fn = { name: cb.name, body: cb.toString() };
+				logger.error({
+					msg: "⚠️ Memoize: unstable_cache failed, using uncached function",
+					error,
+					args,
+					fn,
+					additionalCacheKey,
+				});
+				Sentry.captureException(error, {
+					extra: { args, fn, additionalCacheKey },
```

Comment:
We don't actually have any info to identify the original calling function here when this is being called with autocache (`cb` will be an anonymous function wrapping the actual query fn) , but the `additionalCacheKey` should include the compiled sql which will help us narrow it down. if we really wanted to get the name, we'd need to create an error in `autoCache`, parse the calling fn name from the stack trace, then pass it to `directAutoOutput` and set it as a sentry tag there. felt like overkill for now